### PR TITLE
Cxf-soap tests: Refactor and split

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/4171

### CODE REVIEW

**Comment by JiriOndrusek** [comment]
Replaces https://github.com/apache/camel-quarkus/pull/4172

File: integration-test-groups/cxf-soap/cxf-soap-client/README.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+All clients access services running in containers.
+
+== Maintenenance notes
+
+=== `CalculatorService.wsdl`
+
+`src/main/resources/wsdl/CalculatorService.wsdl` is a static copy of the WSDL served by the testing container.
+It would be too complicated to start the container before running the plugin, so we rather keep the static copy.
+
+There is `io.quarkiverse.cxf.client.it.CxfClientTest.wsdlUpToDate()` to ensure that it is up to date.
```
Comment (ppalaga) [inline integration-test-groups/cxf-soap/cxf-soap-client/README.adoc:13]:
```suggestion
There is `org.apache.camel.quarkus.component.cxf.soap.it.CxfSoapClientTest.wsdlUpToDate()` to ensure that it is up to date.
```

File: integration-test-groups/cxf-soap/cxf-soap-server/src/test/java/org/apache/camel/quarkus/component/cxf/soap/it/CxfSoapServiceTest.java [hunk 2]
```diff
@@ -38,7 +38,7 @@ public void simpleSoapService() {
         final HelloPortType helloPort = service.getHelloPort();
         String endpointURL = getServerUrl() + "/soapservice/hello";
         ((BindingProvider) helloPort).getRequestContext().put(BindingProvider.ENDPOINT_ADDRESS_PROPERTY, endpointURL);
-        Assertions.assertThat(helloPort.hello("World")).isEqualTo("Hello World from CXF service");
```
Comment (ppalaga) [inline integration-test-groups/cxf-soap/cxf-soap-server/src/test/java/org/apache/camel/quarkus/component/cxf/soap/it/CxfSoapServiceTest.java:41]: We could have kept using assert-j it tends to have nicer error messages. Not worth bothering with it right now, let's keep it as you have it.

File: integration-test-groups/cxf-soap/cxf-soap-ws-security/README.adoc [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+All clients access services running in containers.
+
+== Maintenenance notes
+
+=== `CalculatorService.wsdl`
+
+`src/main/resources/wsdl/WssCalculatorService.wsdl` is a static copy of the WSDL served by the testing container.
+It would be too complicated to start the container before running the plugin, so we rather keep the static copy.
+
+There is `io.quarkiverse.cxf.client.it.CxfClientTest.wsdlUpToDate()` to ensure that it is up to date.
```
Comment (ppalaga) [inline integration-test-groups/cxf-soap/cxf-soap-ws-security/README.adoc:13]:
```suggestion
There is `org.apache.camel.quarkus.component.cxf.soap.it.CxfSoapClientTest.wsdlUpToDate()` to ensure that it is up to date.
```

**Review by ppalaga — COMMENTED** [review]
Great work, thanks @JiriOndrusek. I have added some minor comments inline. Besides that, we'd need the `integration-tests/cxf-soap-grouped` module, much like we have for the other groups, e.g. [AWS2](https://github.com/apache/camel-quarkus/tree/main/integration-tests/aws2-grouped)

Rebasing on top of the recent upstream/main would be great too.

File: integration-test-groups/cxf-soap/cxf-soap-ws-security/src/main/java/org/apache/camel/quarkus/component/cxf/soap/it/CxfSoapResource.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    ProducerTemplate producerTemplate;
+
+    @Path("/modulo")
+    @POST
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response modulo(@QueryParam("a") int a,
+            @QueryParam("b") int b)
+            throws Exception {
+        Modulo modulo = new Modulo();
```
Comment (llowinge) [inline integration-test-groups/cxf-soap/cxf-soap-ws-security/src/main/java/org/apache/camel/quarkus/component/cxf/soap/it/CxfSoapResource.java:48]: @JiriOndrusek Where is this `module` being used ?

File: integration-test-groups/cxf-soap/cxf-soap-client/README.adoc (hunk 1 above)
Comment (JiriOndrusek) [reply integration-test-groups/cxf-soap/cxf-soap-client/README.adoc]: fixed

File: integration-test-groups/cxf-soap/cxf-soap-ws-security/README.adoc (hunk 3 above)
Comment (JiriOndrusek) [reply integration-test-groups/cxf-soap/cxf-soap-ws-security/README.adoc]: fixed

File: integration-test-groups/cxf-soap/cxf-soap-ws-security/src/main/java/org/apache/camel/quarkus/component/cxf/soap/it/CxfSoapResource.java (hunk 4 above)
Comment (JiriOndrusek) [reply integration-test-groups/cxf-soap/cxf-soap-ws-security/src/main/java/org/apache/camel/quarkus/component/cxf/soap/it/CxfSoapResource.java]: removed

**Comment by JiriOndrusek** [comment]
`integration-tests/cxf-soap-grouped` was added

File: integration-test-groups/cxf-soap/cxf-soap-server/src/test/java/org/apache/camel/quarkus/component/cxf/soap/it/CxfSoapServiceTest.java (hunk 2 above)
Comment (JiriOndrusek) [reply integration-test-groups/cxf-soap/cxf-soap-server/src/test/java/org/apache/camel/quarkus/component/cxf/soap/it/CxfSoapServiceTest.java]: thanks for noticing!

**Comment by JiriOndrusek** [comment]
@ppalaga I noticed a native error thrown for the test of `cxf-soap-ws-security`. It is:

```
com.oracle.svm.core.util.UserError$UserException: Classes that should be initialized at run time got initialized during image building:
 org.ehcache.sizeof.impl.JvmInformation$15 the class was requested to be initialized at run time (subtype of org.ehcache.sizeof.impl.JvmInformation). To see why org.ehcache.sizeof.impl.JvmInformation$15 got initialized use --trace-class-initialization=org.ehcache.sizeof.impl.JvmInformation$15
org.ehcache.sizeof.impl.JvmInformation the class was requested to be initialized at run time (from feature io.quarkus.runner.Feature.beforeAnalysis with 'JvmInformation.class'). org.ehcache.sizeof.impl.AgentSizeOf caused initialization of this class with the following trace: 
	at org.ehcache.sizeof.impl.JvmInformation.<clinit>(JvmInformation.java:40)
	at org.ehcache.sizeof.impl.AgentLoader.warnIfOSX(AgentLoader.java:159)
	at org.ehcache.sizeof.impl.AgentLoader.loadAgent(AgentLoader.java:130)
	at org.ehcache.sizeof.impl.AgentSizeOf.<clinit>(AgentSizeOf.java:42)

```

Did you encounter similar problem, i.e. during upgrade of `quarkus-cxf`?

**Comment by ppalaga** [comment]
@JiriOndrusek could you please run `mvn process-resources -Pformat -N` from the root folder and amend?

**Comment by JiriOndrusek** [comment]
Changes are formatted by `mvn process-resources -Pformat -N`

**Comment by ppalaga** [comment]
Two more nitpicks: 

* `cxf-codegen-plugin` should be added to `pluginManagement` in build-parent with `${cxf.version}` and `<version>${cxf.version}</version>` can then be removed from `integration-test-groups/cxf-soap/cxf-soap-server/pom.xml`
* The `cxf-xjc-ts` dependency we have in integration-test-groups/cxf-soap/cxf-soap-server/pom.xml should also be present in integration-tests/cxf-soap-grouped/pom.xml

**Comment by ppalaga** [comment]
@JiriOndrusek cxf-soap-ws-security works for me in native mode with rev 5cc40ca

**Comment by ppalaga** [comment]
> Two more nitpicks:
> 
> * `cxf-codegen-plugin` should be added to `pluginManagement` in build-parent with `${cxf.version}` and `<version>${cxf.version}</version>` can then be removed from `integration-test-groups/cxf-soap/cxf-soap-server/pom.xml`
> * The `cxf-xjc-ts` dependency we have in integration-test-groups/cxf-soap/cxf-soap-server/pom.xml should also be present in integration-tests/cxf-soap-grouped/pom.xml

I have done that in https://github.com/apache/camel-quarkus/pull/4195 so that it builds overnight and @llowinge can add his stuff on top of it.

**Pull request closed without merging**